### PR TITLE
Add codecov report

### PR DESCRIPTION
**Pull request by mhagen-vmware**
@jooskim @jak-atx It looks like you currently only run unit tests after merges to master, so I have limited the plugin to that as well, but you should really consider breaking up your Makefile maybe and allowing unit test execution on PRs.

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -81,6 +81,15 @@ pipeline:
       branch: [ refs/tags/* ]
       status: success
 
+  report-coverage:
+     image: robertstettner/drone-codecov
+     token: ${CODECOV_TOKEN}
+     files:
+       - h5c/vic-service/unittest-results/*.xml
```
Comment (jooskim) [inline .drone.yml]: @mhagen-vmware We run unit test coverage for every build. And I see your only adding coverage report for our Java service portion but not including UI coverage, which can be found under `h5c/vic/src/vic-webapp/coverage`

File: .drone.yml [hunk 2]
```diff
@@ -28,9 +28,6 @@ pipeline:
       GS_PRIVATE_KEY: ${GS_PRIVATE_KEY}
     commands:
       - make vic-ui-plugins
-    when:
-      repo: vmware/vic-ui
-      branch: [ master, releases/*, refs/tags/* ]
```
Comment (mhagen-vmware) [inline .drone.yml]: @jooskim this line here says that the tests are definitely not being run on PRs (and of course I can look at the build output of a PR build and see that they are not being run) but if I remove it like this then they will be from now on.

File: .drone.yml [hunk 3]
```diff
@@ -81,6 +78,12 @@ pipeline:
       branch: [ refs/tags/* ]
       status: success
 
+  report-coverage:
+     image: robertstettner/drone-codecov
+     token: ${CODECOV_TOKEN}
+     files:
+       - h5c/vic/src/vic-webapp/coverage/lcov.info
```
Comment (mhagen-vmware) [inline .drone.yml]: @jooskim AFAICT, the java tests do not (have never?) generated any coverage data. 
```
run-test:
    [mkdir] Created dir: /go/src/github.com/vmware/vic-ui/h5c/vic-service/unittest-results
    [junit] Running com.vmware.vic.test.ConfigLoaderTest
    [junit] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.077 sec
    [junit] Running com.vmware.vic.test.ContainerVmTest
    [junit] Tests run: 10, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.255 sec
    [junit] Running com.vmware.vic.test.ModelObjectUriResolverTest
    [junit] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.077 sec
    [junit] Running com.vmware.vic.test.RootTest
    [junit] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.077 sec
    [junit] Running com.vmware.vic.test.VirtualContainerHostVmTest
    [junit] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.249 sec
    [junit] Running com.vmware.vic.test.VmQueryResultTest
    [junit] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.247 sec
```
In order to actually generate coverage data for them, something like jacoco would need to be included in your ant xml file...

Comment (mhagen-vmware) [reply .drone.yml]: From everything I can find, this just runs the unit tests without coverage generation:
https://github.com/vmware/vic-ui/blob/master/h5c/vic-service/build-java.xml#L78

File: .drone.yml (hunk 2 above)
Comment (mhagen-vmware) [reply .drone.yml]: figured it out, sorry.

**Comment by mhagen-vmware** [comment]
From what I can tell comparing the two results ( the build results and the codecov results ) it looks like they are generally the same results except that the build results includes the partial line hits and codecov does not include it.  Thus the diff between 73% and 63%.

**Comment by jooskim** [comment]
@mhagen-vmware If you mean the build results from Java, you are correct we haven't generated any coverage for the Java portion. However the coverage number you see in the build log is coming from the TypeScript portion NOT Java.

**Comment by mhagen-vmware** [comment]
Yea I know it is coming from ts not the Java... I guess really not understanding what you are requesting at this point. You said you wanted me to add something else to this?  But the directory you pointed me at before does not have any coverage report generated.

**Comment by jooskim** [comment]
Oh wait... I thought until now that you've been referring to `vic-service/unittest-results`. You are already getting the right `lcov.info` file for the Angular app. It's the Java portion that does not generate its own coverage report. Sorry for the confusion @mhagen-vmware

**Pull request merged**